src/main/java/lotto/WinningResult.java

```diff
@@ -1,40 +1,36 @@
 package lotto;
 
 import java.util.Arrays;
+import java.util.function.Predicate;
 
 public enum WinningResult {
-    NOT_MATCH(Score.of(0), 0),
-    MATCH_ONE(Score.of(1), 0),
-    MATCH_TWO(Score.of(2), 0),
-    WIN_FOURTH(Score.of(3), 5000),
-    WIN_THIRD(Score.of(4), 50000),
-    WIN_SECOND(Score.of(5), 1500000),
-    WIN_SECOND_BONUS(Score.of(5.5), 30000000),
-    WIN_FIRST(Score.of(6), 2000000000);
-
-    private final Score matchScore;
-    private final int winningPrice;
-
-    WinningResult(Score matchScore, int winningResult) {
-        this.matchScore = matchScore;
+    NOT_MATCH(score -> score.isBetween(0, 1), Money.of(0)),
+    MATCH_ONE(score -> score.isBetween(1, 2), Money.of(0)),
+    MATCH_TWO(score -> score.isBetween(2, 3), Money.of(0)),
+    WIN_FOURTH(score -> score.isBetween(3, 4), Money.of(5000)),
+    WIN_THIRD(score -> score.isBetween(4, 5), Money.of(50000)),
+    WIN_SECOND(score -> score.equals(Score.of(5)), Money.of(1500000)),
+    WIN_SECOND_BONUS(score -> score.isBetween(5.5, 6), Money.of(30000000)),
+    WIN_FIRST(score -> score.isBetween(6, 7), Money.of(2000000000));
+    private Predicate<Score> matchScoreRange;
+    private final Money winningPrice;
```

Comment:
`Predicate`  활용을 잘해주신 것 같아요 👍👍👍
```suggestion
    WIN_FIRST(score -> score.isBetween(6, 7), Money.of(2000000000));
    private final Predicate<Score> matchScoreRange;
    private final Money winningPrice;
```

Reply:
고민이 많았던 부분인데 감사합니다 ! 
불변 선언을 놓쳤네요 ! 
피드백 반영하였습니다 🙇🏻‍♂️